### DISCUSSION TITLE
s-lang package for mingw

### DISCUSSION THREAD

**Discussion by demosthenesk** [msys2/MINGW-packages#13394] [Ideas]
Hello, 
i want to ask if do you plan to include in packages the s-lang library 
https://github.com/hankem/S-Lang 
? 
Thanks Dim !

**Comment by Biswa96** [comment]
The build process is not straight for Windows platform but it supports mingw toolchain. Let see how things goes...

**Comment by demosthenesk** [comment]
I tried to build with mingw64 but i get make errors. :(

**Reply by Biswa96** [reply]
There are two ways to build it. 1. The configure script but it does not support mingw toolchain 2. batch script which is not compatible with mingw environment and directories.

**Comment by demosthenesk** [comment]
so... there is no way to build it properly in msys2 mingw...!

**Reply by Biswa96** [reply]
there is no way to build it properly in msys2 mingw **yet**. But it can be modified to support mingw toolchain. I am poking random stuff. If anything works I would ask it in upstream.

**Reply by demosthenesk** [reply]
thank you !

**Reply by revelator** [reply]
hmm i seem to remember an old port to the mingw.org compiler, might be worth looking into even if the code has evolved.

**Comment by Biswa96** [comment]
Do you need all the modules listed here https://archlinux.org/packages/extra/x86_64/slang/ ?

**Reply by demosthenesk** [reply]
> Do you need all the modules listed here https://archlinux.org/packages/extra/x86_64/slang/ ?

i need only libnewt and libnewt-devel

**Reply by Biswa96** [reply]
libnewt does not support native Windows platform. See the following compiler error:

```
newt.c:7:10: fatal error: sys/signal.h: No such file or directory
    7 | #include <sys/signal.h>
      |          ^~~~~~~~~~~~~~
compilation terminated.
```

**Reply by demosthenesk** [reply]
newt is a TUI for linux terminals.
For windows needs an emulation layer to work like cygwin.
i wanted s-lang in order to compile newt but as i see it can't be compiled native in windows... !
Anyway thanks for your time.
